src/Common/CreditCard.php

```diff
@@ -255,7 +255,7 @@ public function validate()
             throw new InvalidCreditCardException('Card number is invalid');
         }
 
-        if (!is_null($this->getNumber()) && !preg_match('/^\d{12,19}$/i', $this->getNumber())) {
+        if ($this->getNumber() !== '' && !preg_match('/^\d{12,19}$/i', $this->getNumber())) {
```

Comment:
Is this always a string? Cause the check is changed here. Perhaps the phpdocs for the getNumber are not correct instead?

Reply:
I really do not know, I'm not a user and this is an inspirational PR to promote PHPStan :)